.github/workflows/test-kfp.yml

```diff
@@ -0,0 +1,174 @@
+name: Test/build KFP
+
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - "dev"
+            - "releases/**"
+        tags:
+            - "*"
+        paths:
+            - "kfp/**"
+            - "!**.md"
+            - "!**/doc/**"
+            - "!**/.gitignore"
+    pull_request:
+        branches:
+            - "dev"
+            - "releases/**"
+        paths:
+            - "kfp/**"
```

Comment:
kfp workflow test should actually be triggered every time a change is made in a transform, its dependent libs or kfp components... for example, to check that a change in transform parameters is ok so I would not trigger it only in `kfp` directory change. It is currently choosing the transform to run randomly and its not ideal, it might be worth checking if the kfp test could be integrated in each of the transform workflow.

Reply:
If we want randomness, I see no reason to impose this on a transform.  If we want every transform to always run its kfp test, then kfp_ray/Makefile needs to implement the test targets.  However, this would run all kfp transform tests on any change to the core dpk libs.  This might be too much.  I suggest instead we test kfp on a random transform whenever kfp/** changes.

Reply:
And, we could add a target for transform/*/*/kfp_ray/** to trigger a test there if the kfp_ray files change.